fs/layer/layer.go

```diff
@@ -322,7 +322,7 @@ func (r *Resolver) Resolve(ctx context.Context, hosts source.RegistryHosts, refs
 			commonmetrics.MeasureLatencyInMilliseconds(commonmetrics.InitMetadataStore, desc.Digest, start)
 		},
 	}
-	meta, err := r.metadataStore(sr, ztoc, append(metadataOpts, metadata.WithTelemetry(&telemetry))...)
+	meta, err := r.metadataStore(sr, ztoc.TOC, append(metadataOpts, metadata.WithTelemetry(&telemetry))...)
```

Comment:
Since `TOC` is embedded in the `Ztoc`, do we actually need to change these call sites? Do you just want to be more clear?

Reply:
I think it make it more clear that `metadata` only depends on `TOC`. Before looking into the metadata code, I was not sure know how `Ztoc` as a whole is used in it.

Reply:
also in general I feel it's better to avoid pointer args unless (1) we want to change that object; (2) the object is too large and we want to reduce data copy. This can help reduce accidentally changing the object.